tools/rapids-retry

```diff
@@ -26,25 +29,68 @@
 #                          seconds, to wait between retries.
 #                          Default is a 10 second sleep
 #
+# Additional options:
+#
+# --quiet                - when specified as the first argument after the command,
+#                          captures the command's output instead of sending it to stdout.
+#                          Logger messages are redirected to stderr.
+#                          Returns the output of the command on success.
+#
 function rapids-retry {
+    # Handle the --quiet flag if it's the first argument
+    quiet=false
+    if [[ "$1" == "--quiet" ]]; then
+        quiet=true
+        shift
+    fi
     command=$1
     shift
     max_retries=${RAPIDS_RETRY_MAX:=3}
     retries=0
     sleep_interval=${RAPIDS_RETRY_SLEEP:=10}
 
-    ${command} "$@"
-    retcode=$?
-    while (( retcode != 0 )) && \
-          (( retries < max_retries )); do
-      ((retries++))
-      rapids-logger "rapids-retry: retry ${retries} of ${max_retries} | exit code: (${retcode}) -> sleeping for ${sleep_interval} seconds..."
-      sleep "${sleep_interval}"
-      rapids-logger "rapids-retry: sleep done -> retrying..."
-
-      ${command} "$@"
-      retcode=$?
+    # Set up file descriptor 3 - point to stderr if quiet, otherwise stdout
+    if [[ "$quiet" == "true" ]]; then
+        exec 3>&2  # FD 3 points to stderr
+    else
+        exec 3>&1  # FD 3 points to stdout
+    fi
```

Comment:
In my opinion, adding new file descriptors and doing redirects like this is way too much complexity for this purpose.

In https://github.com/rapidsai/gha-tools/pull/164#pullrequestreview-2807558274, I was envisioning something much simpler, just guarding the 2 `rapids-logger` calls based on whether or not `--quiet` was passed.

```text
if [[ "${quiet}" == "true" ]]; then
    rapids-logger ...
fi
```

Reply:
I would agree with this. I don't think FD 3 is needed here.

Reply:
Updated in 0685a39.